synfig-studio/src/gui/dialogs/canvasproperties.h

```diff
@@ -43,39 +55,137 @@
 /* === C L A S S E S & S T R U C T S ======================================= */
 
 namespace Gtk { class TreeView; };
-namespace synfigapp { class CanvasInterface; };
 
 namespace studio
 {
 
 class CanvasProperties  :  public Gtk::Dialog
 {
+	Glib::RefPtr<Gtk::Builder> canvas_properties;
+
 	etl::handle<synfigapp::CanvasInterface> canvas_interface_;
-	Widget_RendDesc widget_rend_desc;
-	Gtk::Entry entry_id;
-	Gtk::Entry entry_name;
-	Gtk::Entry entry_description;
+	synfig::RendDesc rend_desc_;
+
+	Gtk::Box *outer_box;
+	Gtk::Box *inner_box;
+
+	// Canvas Info
+	Gtk::Entry *entry_id;
+	Gtk::Entry *entry_name;
+	Gtk::Entry *entry_desc;
+
+	// Image
+	Gtk::Label *ratio_label;
+
+	Gtk::SpinButton *entry_width;
+	Gtk::SpinButton *entry_height;
+	Gtk::SpinButton *entry_xres;
+	Gtk::SpinButton *entry_yres;
+	Gtk::SpinButton *entry_phy_w;
+	Gtk::SpinButton *entry_phy_h;
+	Gtk::SpinButton *entry_span;
+
+	Gtk::Image *toggle_wh_img;
+	Gtk::Image *toggle_res_img;
+
+	Gtk::ToggleButton *toggle_wh_r;
+	Gtk::ToggleButton *toggle_res_r;
+
+	const std::string tooltip_wh_on;
+	const std::string tooltip_wh_off;
+	const std::string tooltip_res_on;
+	const std::string tooltip_res_off;
+
+	Widget_Vector *entry_tl;
+	Widget_Vector *entry_br;
+
+	// Time
+	Gtk::SpinButton *entry_fps;
+
+	Widget_Time *entry_start;
+	Widget_Time *entry_end;
+	Widget_Time *entry_dur;
+
+	// Gamma Correction
+	Gtk::SpinButton *entry_gamma_r;
+	Gtk::SpinButton *entry_gamma_g;
+	Gtk::SpinButton *entry_gamma_b;
+
+	Gtk::Scale *scale_gamma_r;
+	Gtk::Scale *scale_gamma_g;
+	Gtk::Scale *scale_gamma_b;
+
+	studio::Widget_GammaPattern *gamma_pattern;
+
+	// Other
+	Gtk::CheckButton *toggle_pxl_w;
+	Gtk::CheckButton *toggle_pxl_h;
+	Gtk::CheckButton *toggle_pxl_a;
+	Gtk::CheckButton *toggle_img_w;
+	Gtk::CheckButton *toggle_img_h;
+	Gtk::CheckButton *toggle_img_a;
+	Gtk::CheckButton *toggle_img_s;
+
+	Widget_Vector *entry_focus;
 
 	bool dirty_rend_desc;
 
+	mutable int update_lock;
+
+	struct UpdateLock
+	{
+		int &locked;
+		UpdateLock(int &locked):locked(locked){locked++;}
+		~UpdateLock(){locked--;}
+	};
+
 	//Gtk::TreeView* meta_data_tree_view;
 	//void on_button_meta_data_add();
 	//void on_button_meta_data_delete();
 
 public:
-	CanvasProperties(Gtk::Window& parent,etl::handle<synfigapp::CanvasInterface> canvas_interface);
+	CanvasProperties(Gtk::Window &, etl::handle<synfigapp::CanvasInterface>);
```

Comment:
no problem at all, but why bother to remove the parameter names here?

Reply:
My habit. I prefer the header file as a design template, so in practice, I only need to know what types the arguments being passed are.

Let's say I use parameter names here too. But when I implement the constructor in .cpp file the IDE doesn't autocomplete those names for me. So I'm typing the same thing twice both for declaration and definition file.

Honestly, it's just my taste. But if you prefer to keep this file consistent with other files, I could bring back those parameter name again. Conforming to project standard is better than having multiple code writing styles from different contributors

Reply:
I just think it's a good "autodoc"/tip for those who try to use the method :)

Reply:
Alright, bringing back those names :)